core/src/main/java/org/elasticsearch/search/aggregations/metrics/percentiles/hdr/HDRPercentileRanksAggregator.java

```diff
@@ -76,21 +85,43 @@ public double metric(String name, long bucketOrd) {
 
     public static class Factory extends ValuesSourceAggregatorFactory.LeafOnly<ValuesSource.Numeric> {
 
-        private final double[] values;
-        private final int numberOfSignificantValueDigits;
-        private final boolean keyed;
+        private double[] values;
+        private int numberOfSignificantValueDigits = 3;
+        private boolean keyed = false;
 
-        public Factory(String name, ValuesSourceParser.Input<ValuesSource.Numeric> valuesSourceInput, double[] values,
-                int numberOfSignificantValueDigits, boolean keyed) {
-            super(name, InternalHDRPercentiles.TYPE.name(), valuesSourceInput);
+        public Factory(String name) {
+            super(name, InternalHDRPercentileRanks.TYPE, ValuesSourceType.NUMERIC, ValueType.NUMERIC);
+        }
+
+        public void values(double[] values) {
             this.values = values;
-            this.numberOfSignificantValueDigits = numberOfSignificantValueDigits;
+        }
+
+        public double[] values() {
+            return values;
+        }
+
+        public void keyed(boolean keyed) {
             this.keyed = keyed;
         }
 
+        public boolean keyed() {
+            return keyed;
+        }
+
+        public void numberOfSignificantValueDigits(int numberOfSignificantValueDigits) {
+            this.numberOfSignificantValueDigits = numberOfSignificantValueDigits;
+        }
+
+        public int numberOfSignificantValueDigits() {
+            return numberOfSignificantValueDigits;
+        }
+
         @Override
         protected Aggregator createUnmapped(AggregationContext aggregationContext, Aggregator parent,
                 List<PipelineAggregator> pipelineAggregators, Map<String, Object> metaData) throws IOException {
+            double[] values = Arrays.copyOf(this.values, this.values.length);
+            Arrays.sort(values);
```

Comment:
For the record, I'm asking because I expect the setter to be called once while these create methods can be called _many_ times